form-flow-valtimo/src/test/kotlin/com/ritense/valtimo/formflow/common/ValtimoFormFlowIntTest.kt

```diff
@@ -0,0 +1,172 @@
+
+@Transactional
+class ValtimoFormFlowIntTest : BaseIntegrationTest() {
+
+    @Autowired
+    lateinit var processDocumentService: ProcessDocumentService
+
+    @Autowired
+    lateinit var formAssociationService: FormAssociationService
+
+    @Autowired
+    lateinit var formFlowInstanceRepository: FormFlowInstanceRepository
+
+    @Autowired
+    lateinit var taskService: TaskService
+
+    @Autowired
+    lateinit var formFlowService: FormFlowService
+
+    @Autowired
+    lateinit var formFlowResource: FormFlowResource
+
+    @Autowired
+    lateinit var formFlowDeploymentService: FormFlowDeploymentService
+
+    @Autowired
+    lateinit var historyService: HistoryService
+
+    @Test
+    fun `should completeTask`() {
+        deployFormFlow(onComplete = "\${valtimoFormFlow.completeTask(additionalProperties)}")
+        linkFormFlowToUserTask()
+        val documentAndProcess = newDocumentAndStartProcess()
+        val processInstanceId = documentAndProcess.resultingProcessInstanceId().get()
+        val formFlowInstance = getFormFlowInstances(processInstanceId).single()
+
+        formFlowStepComplete(formFlowInstance)
+
+        assertEquals(0, taskService.createTaskQuery().processInstanceId(processInstanceId.toString()).count())
+    }
+
+    @Test
+    fun `should completeTask and save submission in document`() {
+        deployFormFlow(onComplete = "\${valtimoFormFlow.completeTask(additionalProperties, step.submissionData)}")
+        linkFormFlowToUserTask()
+        val documentAndProcess = newDocumentAndStartProcess()
+        val processInstanceId = documentAndProcess.resultingProcessInstanceId().get()
+        val formFlowInstance = getFormFlowInstances(processInstanceId).single()
+
+        formFlowStepComplete(formFlowInstance, submission = """{"firstName":"John"}""")
+
+        assertEquals(0, taskService.createTaskQuery().processInstanceId(processInstanceId.toString()).count())
+        assertEquals(
+            """{"submission":{"firstName":"John"}}""",
+            documentAndProcess.resultingDocument().get().content().asJson().toString()
+        )
+    }
+
+    @Test
+    fun `should completeTask and save submission in document and process`() {
+        deployFormFlow(onComplete = "\${valtimoFormFlow.completeTask(additionalProperties, step.submissionData, {'doc:/address/streetName':'/street', 'pv:approved':'/approval'})}")
+        linkFormFlowToUserTask()
+        val documentAndProcess = newDocumentAndStartProcess()
+        val processInstanceId = documentAndProcess.resultingProcessInstanceId().get()
+        val formFlowInstance = getFormFlowInstances(processInstanceId).single()
+
+        formFlowStepComplete(formFlowInstance, submission = """{"street":"Funenpark","approval":true}""")
+
+        assertEquals(0, taskService.createTaskQuery().processInstanceId(processInstanceId.toString()).count())
+        assertEquals(
+            """{"address":{"streetName":"Funenpark"}}""",
```

Comment:
shouldn't this json also have the property "approval:true"?

Reply:
Not in this test no. Because I created some setting that can decide where to save the data.
In this test the setting is `'pv:approved':'/approval'` Which translates to 'save the approval in a process variable'